tests/integration_tests/cmd/test_status.py

```diff
@@ -82,3 +82,57 @@ def test_status_json_errors(client):
     assert "Invalid cloud-config provided" in json.loads(status_json)[
         "recoverable_errors"
     ].get("WARNING").pop(0)
+
+
+EARLY_BOOT_WAIT_USER_DATA = """\
+#cloud-config
+runcmd: [systemctl enable before-cloud-init-local.service]
+write_files:
+- path: /waitoncloudinit.sh
+  permissions: '0755'
+  content: |
+    #!/bin/sh
+    [ -f /var/lib/cloud/data/status.json ] && touch /$1.start-hasstatusjson || touch /$1.start-nostatusjson
+    cloud-init status --wait --long > $1
+- path: /lib/systemd/system/before-cloud-init-local.service
+  permissions: '0644'
+  content: |
+    [Unit]
+    Description=BEFORE cloud-init local
+    DefaultDependencies=no
+    Before=cloud-init-local.service
+    Before=shutdown.target
+    Before=sysinit.target
+    Conflicts=shutdown.target
+    RequiresMountsFor=/var/lib/cloud
+
+    [Service]
+    Type=simple
+    ExecStart=/waitoncloudinit.sh /before-local
+    RemainAfterExit=yes
+    TimeoutSec=0
+
+    # Output needs to appear in instance console output
+    StandardOutput=journal+console
+
+    [Install]
+    WantedBy=cloud-init.target
+"""  # noqa: E501
+
+
+@pytest.mark.user_data(EARLY_BOOT_WAIT_USER_DATA)
+@pytest.mark.lxd_use_exec
```

Comment:
We should add a skip if not LXD here. Any other platform we need to wait for ssh access to get in and run the command.

Reply:
added.